tests/examples/single-select.test.js

```diff
@@ -0,0 +1,127 @@
+import "jest";
+import { create, parameterized, Tree, use } from "microstates";
+import { foldl } from "funcadelic";
+import values from '../../src/values';
+class Color {}
+
+function selectMiddleware(next, tree) {
+  return (microstate, transition, args) => {
+    let focus = Tree.from(microstate);
+
+    let isChildFocused = foldl((isChild, child) => {
+      return isChild ? true : focus.is(child);
+    }, false, values(tree.children));
+
+    if (transition.name === 'choose' && isChildFocused) {
+      return tree.microstate.map(selectable => {
+        let current = Tree.from(selectable);
+        if (current.is(focus)) {
+          return current.microstate.isSelected.set(true);
```

Comment:
I feel as though this should be redundant. The transition happened at the site of the choice. We shouldn't have to deal with that case by default. What we're wanting to do is to reconcile the consequences higher up the tree with that transition. I don't fully understand this middleware yet, but I think it's a fault of our system if "pushing forward" a transition that has already been sanctioned is going to be a hard requirement.

Reply:
Agree, but our middleware is not complete. I can explain what's missing in person - requires diagrams.